src/dec/valkyria/odn_archive_decoder.cc

```diff
@@ -99,16 +109,16 @@ static void fill_sizes(
     last_entry->size = input_stream.size() - last_entry->offset;
 }
 
-static std::unique_ptr<dec::ArchiveMeta> read_meta_v1(
+static void read_meta_v1(
+    dec::ArchiveMeta *meta,
     io::BaseByteStream &input_stream)
 {
-    auto meta = std::make_unique<dec::ArchiveMeta>();
     while (true)
     {
         auto entry = std::make_unique<dec::PlainArchiveEntry>();
         entry->path = input_stream.read(8).str();
         entry->offset = hex_to_int(input_stream.read(8));
-        if (entry->path.str().substr(0, 4) == "END_")
+        if (entry->path == "END_ffff" || entry->path == "ffffffff")
```

Comment:
There are apparently several variants for the terminator. I added the variation I saw after decryption of the archive.